### PR TITLE
SONARJAVA-547 SONARJAVA-1075 Added missing tokens and remove usage of ast node

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 1]
```diff
@@ -99,18 +106,23 @@ public MethodTreeImpl(
...
   }
 
-  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName) {
+  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName, SyntaxToken semicolonToken) {
     Preconditions.checkState(this.simpleName == null);
     this.returnType = returnType;
     this.simpleName = simpleName;
+    this.semicolonToken = semicolonToken;
 
     prependChildren((AstNode) returnType, (AstNode) simpleName);
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: MAJORorg/sonar/java/model/declaration/MethodTreeImpl must not use com/sonar/sslr/api/AstNode[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)

File: java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java [hunk 2]
```diff
@@ -41,43 +64,186 @@
...
-  }
+    @Override
+    public void visitCatch(CatchTree tree) {
+      scan(tree.block());
+    }
+
+    @Override
+    public void visitMethod(MethodTree tree) {
+      SyntaxToken semicolonToken = tree.semicolonToken();
+      if (semicolonToken != null) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java]: NewCRITICALRedundant nullcheck of semicolonToken, which is known to be non-null in org.sonar.java.checks.RightCurlyBraceStartLineCheck$SyntaxTokenFinder.visitMethod(MethodTree)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 critical

Details:
* [Redundant nullcheck of semicolonToken, which is known to be non-null in org.sonar.java.checks.RightCurlyBraceStartLineCheck$SyntaxTokenFinder.visitMethod(MethodTree)](https://github.com/SonarSource/sonar-java/blob/3ad669e96e30d386a44043b531732567359ad904/java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java#L222)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 3]
```diff
@@ -56,14 +57,18 @@
...
   @Nullable
   private final BlockTree block;
+  @Nullable
+  private SyntaxToken semicolonToken;
   private final List<TypeTree> throwsClauses;
   private final SyntaxToken defaultToken;
   private final ExpressionTree defaultValue;
 
-  //FIXME nullable if semantic analysis is not set. Should have a default value.
+  // FIXME nullable if semantic analysis is not set. Should have a default value.
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: MAJORTake the required action to fix the issue indicated by this comment.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 critical

Details:
* [Redundant nullcheck of semicolonToken, which is known to be non-null in org.sonar.java.checks.RightCurlyBraceStartLineCheck$SyntaxTokenFinder.visitMethod(MethodTree)](https://github.com/SonarSource/sonar-java/blob/983497d752925171b2814320fde344baccf09689/java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java#L222)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

File: java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java (hunk 2 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java]: This is a FP. `semicolonToken()` is nullable for methods!

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java (hunk 1 above)
Comment (m-g-sonar) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: A new issue should have appeared on next line too.

**Comment by benzonico** [comment]
LGTM

**Pull request closed without merging**